packages/terra-doc-template/src/DocTemplate.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PropsTable from 'terra-props-table';
+import Markdown from 'terra-markdown';
+import IndexExampleTemplate from './IndexPageTemplate';
+
+const propTypes = {
+  /**
+   * The given component's npm package name
+   */
+  packageName: PropTypes.string,
+  /**
+   * The given component's readme file imported to a string
+   */
+  readme: PropTypes.string,
+  /**
+   * All of the example(s) that will be displayed. An empty array is supported.
+   * ```
+   * title: The title of the example
+   * description: A description of the example to be displayed below the title
+   * example: The example to be displayed
+   * source: The source code of the example
+   * ```
+   */
+  examples: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        title: PropTypes.string,
+        description: PropTypes.string,
+        example: PropTypes.element,
+        source: PropTypes.string,
+      },
+    ),
+  ),
+  /**
+   * All of the props table(s) that will be displayed. An empty array is supported.
+   * ```
+   * componentSrc: The source code of the component
+   * componentName: The name of the component
+   * ```
+   */
+  propsTables: PropTypes.arrayOf(
+    PropTypes.shape(
+      {
+        componentSrc: PropTypes.string,
+        componentName: PropTypes.string,
+      },
+    ),
+  ),
+};
+
+const defaultProps = {
+  packageName: '',
+  readme: '',
+  examples: [],
+  propsTables: [],
+};
+
+const DocTemplate = (props) => {
+  const { packageName, readme, examples, propsTables } = props;
```

Comment:
Do we want to open up `customProps` for consumers to use? Similar to [here](https://github.com/cerner/terra-core/blob/master/packages/terra-status/src/Status.jsx#L20). We'll have to remember to spread them [first](https://github.com/cerner/terra-core/blob/master/packages/terra-status/src/Status.jsx#L21).